dozer-types/src/json_types.rs

```diff
@@ -1,253 +1,88 @@
+    let serde_value = json_value_to_serde_json(value);
+    serde_value.to_string()
 }
 
-impl From<f32> for JsonValue {
-    fn from(f: f32) -> Self {
-        From::from(f as f64)
-    }
+pub(crate) fn json_to_bytes(value: &JsonValue) -> Vec<u8> {
+    rmp_serde::to_vec(value).unwrap()
 }
 
-impl From<f64> for JsonValue {
-    fn from(f: f64) -> Self {
-        JsonValue::Number(OrderedFloat(f))
-    }
+pub(crate) fn json_from_bytes(bytes: &[u8]) -> Result<JsonValue, DeserializationError> {
+    rmp_serde::from_slice(bytes).map_err(Into::into)
 }
 
-impl From<bool> for JsonValue {
-    fn from(f: bool) -> Self {
-        JsonValue::Bool(f)
-    }
+pub fn json_to_bytes_size(value: &JsonValue) -> usize {
+    json_to_bytes(value).len()
 }
 
-impl From<String> for JsonValue {
-    fn from(f: String) -> Self {
-        JsonValue::String(f)
+pub(crate) fn json_cmp(l: &JsonValue, r: &JsonValue) -> std::cmp::Ordering {
+    // Early return in the common case
+    if l.type_() != r.type_() {
+        return l.type_().cmp(&r.type_());
     }
-}
 
-impl<'a> From<&'a str> for JsonValue {
-    fn from(f: &str) -> Self {
-        JsonValue::String(f.to_string())
-    }
-}
-
-impl<'a> From<Cow<'a, str>> for JsonValue {
-    fn from(f: Cow<'a, str>) -> Self {
-        JsonValue::String(f.into_owned())
-    }
-}
-
-impl From<OrderedFloat<f64>> for JsonValue {
-    fn from(f: OrderedFloat<f64>) -> Self {
-        JsonValue::Number(f)
-    }
-}
-
-impl From<BTreeMap<String, JsonValue>> for JsonValue {
-    fn from(f: BTreeMap<String, JsonValue>) -> Self {
-        JsonValue::Object(f)
-    }
-}
-
-impl<T: Into<JsonValue>> From<Vec<T>> for JsonValue {
-    fn from(f: Vec<T>) -> Self {
-        JsonValue::Array(f.into_iter().map(Into::into).collect())
-    }
-}
-
-impl<'a, T: Clone + Into<JsonValue>> From<&'a [T]> for JsonValue {
-    fn from(f: &'a [T]) -> Self {
-        JsonValue::Array(f.iter().cloned().map(Into::into).collect())
-    }
-}
-
-impl<T: Into<JsonValue>> FromIterator<T> for JsonValue {
-    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
-        JsonValue::Array(iter.into_iter().map(Into::into).collect())
-    }
-}
-
-impl<K: Into<String>, V: Into<JsonValue>> FromIterator<(K, V)> for JsonValue {
-    fn from_iter<I: IntoIterator<Item = (K, V)>>(iter: I) -> Self {
-        JsonValue::Object(
-            iter.into_iter()
-                .map(|(k, v)| (k.into(), v.into()))
-                .collect(),
-        )
-    }
-}
-
-impl From<()> for JsonValue {
-    fn from((): ()) -> Self {
-        JsonValue::Null
-    }
-}
-
-impl<T> From<Option<T>> for JsonValue
-where
-    T: Into<JsonValue>,
-{
-    fn from(opt: Option<T>) -> Self {
-        match opt {
-            None => JsonValue::Null,
-            Some(value) => Into::into(value),
+    match (l.destructure_ref(), r.destructure_ref()) {
+        (DestructuredRef::Object(l), DestructuredRef::Object(r)) => {
+            for (l_k, l_v) in l {
```

Comment:
[IObject](https://docs.rs/ijson/latest/ijson/object/struct.IObject.html#) preserves the insertion order, so `Ord` will not be well defined?

Reply:
Good point. I'll iterate left sorted instead, so it's well-defined

Reply:
I did something inspired by `BTreeMap`'s `Ord` implementation and added a test

Reply:
Agreed